### PR TITLE
Hostname check with DNS or not

### PR DESCRIPTION
**Pull request by alexanderkjall**
add an environment variable to control if the hostname check should be performed with a dns resolution or not.

https://github.com/adevinta/vulcan-types/pull/142 needs to be merged and released before this can be evaluated I guess :)

### CODE REVIEW

File: pkg/api/asset.go [hunk 1]
```diff
@@ -67,8 +68,14 @@ func (a Asset) Validate() error {
 
 	switch a.AssetType.Name {
 	case "Hostname":
-		if !types.IsHostname(a.Identifier) {
-			return errors.Validation("Identifier is not a valid Hostname")
+		if os.Getenv("VULCAN_HOSTNAME_VALIDATION_WITH_DNS") == "false" {
```
Comment (manelmontilla) [inline pkg/api/asset.go:71]: We don't directly access env vars in the logic of the program.
In order to add a config option you need to:
1. Add it to the config file that maps env vars to parameters in the [config file](https://github.com/adevinta/vulcan-api/blob/master/config.toml)
I think there isn't any section that makes sense to add this new param, so I would add a new section, something like this:
```toml
[assets]
dns_hostname_validaton = "$DNS_HOSTNAME_VALIDATION"
``` 
2. Map the value in the [run script of the container](https://github.com/adevinta/vulcan-api/blob/master/run.sh) by adding something like this:

```bash
export DNS_HOSTNAME_VALIDATION=${DNS_HOSTNAME_VALIDATION:-true}
```
3. Add the new section in the struct representing the [configuration of the api](https://github.com/adevinta/vulcan-api/blob/master/cmd/vulcan-api/commands.go#L149)

**Review by manelmontilla — CHANGES_REQUESTED** [review]

File: pkg/api/asset.go (hunk 1 above)
Comment (alexanderkjall) [reply pkg/api/asset.go]: Thanks for the review, I updated the branch and tried to apply this feedback.

**Review by manelmontilla — CHANGES_REQUESTED** [review]

File: cmd/vulcan-api/commands.go [hunk 2]
```diff
@@ -146,22 +146,27 @@ type awsCatalogueConfig struct {
 	RetryInterval int    `mapstructure:"retry_interval"`
 }
 
+type dnsHostnameValidation struct {
+	DnsHostnameValidation string `mapstructure:"dns_hostname_validation"`
```
[the remark this answers is not in the document]
Comment (alexanderkjall) [reply cmd/vulcan-api/commands.go:150]: thanks for the link, fixed :)

File: pkg/api/endpoint/assets.go [hunk 3]
```diff
@@ -85,7 +85,7 @@ func makeListAssetsEndpoint(s api.VulcanitoService, logger kitlog.Logger) endpoi
 }
 
 // makeCreateAssetEndpoint returns an endpoint that creates new assets.
-func makeCreateAssetEndpoint(s api.VulcanitoService, logger kitlog.Logger) endpoint.Endpoint {
+func makeCreateAssetEndpoint(s api.VulcanitoService, logger kitlog.Logger, dnsHostnameValidation bool) endpoint.Endpoint {
```
[the remark this answers is not in the document]
Comment (alexanderkjall) [reply pkg/api/endpoint/assets.go:88]: Good suggestion, this should be fixed now.

**Comment by alexanderkjall** [comment]
Any update on this one?

**Comment by alexanderkjall** [comment]
Any update on this one?

**Comment by kozmic** [comment]
Now that the https://github.com/adevinta/vulcan-types/pull/142 has been merged, could this PR be reviewed and merged @manelmontilla ?

**Comment by jesusfcr** [comment]
Fixed and merged https://github.com/adevinta/vulcan-api/pull/317

**Pull request closed without merging**